runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -93,14 +110,38 @@ public Layout getCurrentLayout() {
                     log.warn("Got a wrong epoch exception, updating epoch to {} and "
                             + "invalidate view", we.getCorrectEpoch());
                     runtime.invalidateLayout();
+                } else if (re instanceof NetworkException) {
+                    log.warn("layoutHelper: System seems unavailable", re);
+
+                    runtime.systemDownHandler.run();
+                    runtime.invalidateLayout();
+
+                    try {
+                        Thread.sleep(runtime.retryRate * 1000);
+                    } catch (InterruptedException e) {
+                        log.warn("Interrupted Exception in layout helper.", e);
+                    }
+
                 } else {
                     throw re;
                 }
+                if (rethrowAllExceptions) {
+                    throw new RuntimeException(re);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)